substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/debugentry/Range.java

```diff
@@ -79,6 +83,25 @@ public Range(String fileName, Path filePath, String className, String methodName
         this.primary = primary;
     }
 
+    /*
+     * Create a slightly different copy of a previously constructed range. Because the previous
+     * range was constructed by one of the other constructors, a valid assumption is that all the
+     * strings have previously been inserted int he stringTable, and we can avoid doing that again.
+     */
+    public Range(Range other, int lo, int hi) {
```

Comment:
For that I'd use:
```java
public static Range withLoHi(int lo, int hi) {
    Range that = this.clone();
    that.lo = lo;
    that.hi = hi;
}
```

Reply:
This is much cleaner.  My goal was to avoid modifying Range members after construction (I want to make them final), and to avoid calling uniqueString() multiple times.
While your suggestion doesn't help with my goal of immutability, it is much more readable.
I've adopted it, and may make immutability a separate PR after some discussion.